### PR TITLE
Mocking and CLI test for proxy-status and proxy-config

### PR DESCRIPTION
**Pull request by esnible**
Initial factor to allow mocking and initial Go tests

### CODE REVIEW

File: istioctl/cmd/istioctl/proxystatus.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ import (
 )
 
 var (
+	mockExecClient kubernetes.ExecClient
```
Comment (liamawhite) [inline istioctl/cmd/istioctl/proxystatus.go]: not sure why this is named **mock**ExecClient?

File: istioctl/pkg/kubernetes/client.go [hunk 2]
```diff
@@ -33,12 +33,23 @@ var (
...
+	AllPilotsDiscoveryDo(pilotNamespace, method, path string, body []byte) (map[string][]byte, error)
+}
+
 // Client is a helper wrapper around the Kube RESTClient for istioctl -> Pilot/Envoy/Mesh related things
 type Client struct {
 	Config *rest.Config
 	*rest.RESTClient
 }
 
+type kubeExecClient struct {
```
Comment (liamawhite) [inline istioctl/pkg/kubernetes/client.go]: Why is a second wrapper needed?

File: istioctl/cmd/istioctl/istioctl_test.go [hunk 3]
```diff
@@ -568,3 +576,129 @@ func TestVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestProxyConfig(t *testing.T) {
```
Comment (liamawhite) [inline istioctl/cmd/istioctl/istioctl_test.go]: can we split this out into `proxyconfig_test.go`

File: istioctl/cmd/istioctl/istioctl_test.go [hunk 4]
```diff
@@ -568,3 +576,129 @@ func TestVersion(t *testing.T) {
...
+	}
+
+	for i, c := range cases {
+		t.Run(fmt.Sprintf("case %d %s", i, strings.Join(c.args, " ")), func(t *testing.T) {
+			verifyOutput(t, c)
+		})
+	}
+}
+
+func TestProxyStatus(t *testing.T) {
```
Comment (liamawhite) [inline istioctl/cmd/istioctl/istioctl_test.go]: same but for `proxystatus_test.go`

File: istioctl/cmd/istioctl/istioctl_test.go [hunk 5]
```diff
@@ -568,3 +576,129 @@ func TestVersion(t *testing.T) {
...
+	}
+
+	for i, c := range cases {
+		t.Run(fmt.Sprintf("case %d %s", i, strings.Join(c.args, " ")), func(t *testing.T) {
+			verifyOutput(t, c)
+		})
+	}
+}
+
+func mockClientExecFactoryGenerator(testResults map[string][]byte) func(kubeconfig, configContext string) (kubernetes.ExecClient, error) {
```
Comment (liamawhite) [inline istioctl/cmd/istioctl/istioctl_test.go]: This feels like overkill for just creating a `MockExecClient` struct. This is why I'm not a huge fan of factories in Go.

Not a blocking comment though.

**Comment by rshriram** [comment]
change to release-1.0 branch

**Pull request merged**